src/servers/ZoneServer2016/zonepackethandlers.ts

```diff
@@ -1302,11 +1302,22 @@ export class ZonePacketHandlers {
     if (client.character.tempGodMode) {
       server.setTempGodMode(client, false);
     }
-    client.character.positionUpdate = packet.data;
+    client.character.positionUpdate = {
+      ...client.character.positionUpdate,
+      ...packet.data
+    };
     if (packet.data.flags === 513) {
       // head rotation when in vehicle, client spams this packet every 1ms even if you dont move, disabled for now(it doesnt work anyway)
       return;
     }
+
+    if (packet.data.orientation) {
+      server.fairPlayManager.checkAimVector(
```

Comment:
isn't it too much ? like this packet is really spammed probably at the very least 100/sec on 100+ players. Did you check how much cpu checkAimVector consume at 100+ players ?